transmogrifier/sources/transformer.py

```diff
@@ -242,30 +238,21 @@ def _transform(
             source_record: A single source record.
         """
         if self.record_is_deleted(source_record):
-            source_record_id = self.get_source_record_id(source_record)
-            timdex_record_id = self.get_timdex_record_id(
-                self.source, source_record_id, source_record
-            )
+            timdex_record_id = self.get_timdex_record_id(source_record)
             raise DeletedRecordEvent(timdex_record_id)
-        optional_fields = self.get_optional_fields(source_record)
-        if optional_fields is None:
-            return None
-
-        fields = {
-            **self.get_required_fields(source_record),
-            **optional_fields,
-        }
 
-        fields = self.create_dates_and_locations_from_publishers(fields)
-        fields = self.create_locations_from_spatial_subjects(fields)
+        timdex_record = timdex.TimdexRecord(
+            source=self.source_name,
+            source_link=self.get_source_link(source_record),
+            timdex_record_id=self.get_timdex_record_id(source_record),
+            title=self.get_valid_title(source_record),
+        )
+        for field_name, field_method in self.get_optional_field_methods():
+            setattr(timdex_record, field_name, field_method(source_record))
```

Comment:
What happens when a field method fails?  Seems pretty clear that we don't want to continue with the record, so I think the question is do we want to catch the exception(s) here, or let them bubble up to the `__next__()` method that called this `transform()` method.

Reply:
Maybe it's a combination of the two?

I see in `__next__()` we have this pattern:
```python
source_record = next(self.source_records)
self.processed_record_count += 1
try:
    record = self.transform(source_record)
except DeletedRecordEvent as error:
    self.deleted_records.append(error.timdex_record_id)
    continue
except SkippedRecordEvent:
    self.skipped_record_count += 1
    continue
if not record:
    self.skipped_record_count += 1
    continue
self.transformed_record_count += 1
return record
```

It feels correct that we'd let `DeletedRecordEvent` or `SkippedRecordEvent` bubble up from `transform()` to this level, as it's at the "loop" level of transforming.  

So if a field method has an issue, maybe we want to `try/except` it at the `setattr()` level in `transform()`, log the error (and maybe send metrics in the future...) and then manually raise a `SkipRecord...` exception?

Reply:
> But if a field method has an issue, maybe we want to `try/except` it at the `setattr()` level in `transform()`, log the error (and maybe send metrics in the future...) and then manually raise a `SkipRecord...` exception?

That approach feels good to me and seems to fit with the generalized `SkippedRecordEvent` exception.